### PR TITLE
Flags `x`, `n`, and `s` cannot be used in JavaScript regular expressions

### PR DESCRIPTION
**Pull request by vais**
Flags `x`, `n`, and `s` cannot be used in JavaScript regular expressions, and regular expressions such as /(?ix:foo)/ are invalid in JavaScript

### CODE REVIEW

File: core/regexp/inspect_spec.rb [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 describe "Regexp#inspect" do
   it "returns a formatted string that would eval to the same regexp" do
-    /ab+c/ix.inspect.should == "/ab+c/ix"
-    /a(.)+s/n.inspect.should =~ %r|/a(.)+s/n?|  # Default 'n' may not appear
+    not_supported_on :opal do
+      /ab+c/ix.inspect.should == "/ab+c/ix"
+      /a(.)+s/n.inspect.should =~ %r|/a(.)+s/n?|  # Default 'n' may not appear
```
Comment (eregon) [inline core/regexp/inspect_spec.rb]: Since `n` only makes the Regexp have binary encoding, maybe it could be supported as a simple no-op?

**Comment by eregon** [comment]
You mean `m` right, not `s`?

I think `x` is purely a parse-time transformation, so I suppose it could be supported on Opal but the effort is quite significant. Would supporting it as a no-op be difficult?

**Comment by vais** [comment]
@eregon `m` is a valid regexp flag in JS, `s` is not valid. All you get with JavaSctipt is `g`, `i`, and `m` unfortunately (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/RegExp)

Currently Opal does not do any processing of regexps at all - they are passed straight through to JavaScript. So even a no-op represents a challenge at this time. Perhaps @elia or someone else on @opal/core can correct me if I'm wrong.

I definitely see how these `not_supported_on :opal` blocks are quite obtrusive. In most of the above specs though, the use of the more exotic flags is not necessary for the purpose of the spec. E.g. for the purpose of testing `#inspect`, `/ab+c/ix.inspect.should == "/ab+c/ix"` could as well have been `/ab+c/im.inspect.should == "/ab+c/im"`, and for the purpose of testing `#try_convert`, `Regexp.try_convert(/foo/s).should == /foo/s` could as well be `Regexp.try_convert(/foo/m).should == /foo/m`, etc.

**Comment by vais** [comment]
@eregon also, for what it's worth, this should be the last batch of these regexp-related `not_supported_on: opal`s - these came up as a result of a push to switch from white-listing rubyspecs to black-listing the ones that definitely are a no-go on Opal (https://github.com/opal/opal/pull/936).

In my local Opal repo, the list of specs for regular expressions looks like this now:
```
rubyspec/core/regexp
!rubyspec/core/regexp/named_captures_spec
!rubyspec/core/regexp/names_spec
!rubyspec/core/regexp/encoding_spec
!rubyspec/core/regexp/fixed_encoding_spec
```
Which means run all specs under `rubyspec/core/regexp`, except for the 4 listed starting with a `!`. I have not added this change to https://github.com/opal/opal/pull/936 because it requires this PR to come through first. If this PR cannot be merged, I will just add these 5 lines to the list and be done with it for now. No big deal :smile:
```
!rubyspec/core/regexp/hash_spec
!rubyspec/core/regexp/inspect_spec
!rubyspec/core/regexp/source_spec
!rubyspec/core/regexp/to_s_spec
!rubyspec/core/regexp/try_convert_spec
```

**Comment by eregon** [comment]
@vais I was just asking to know the reasons :grinning: .
On the long term I'd like these to go away but in the meantime it's fine.
I'll merge once RubySpec is stable again.

**Comment by vais** [comment]
Sweet, thanks @eregon!

**Comment by eregon** [comment]
Merged!

**Pull request merged**